gatsby/src/specs/minfraud/2.0/utils.ts

```diff
@@ -0,0 +1,315 @@
+/* eslint-disable security/detect-object-injection */
+import OpenApiParser from '@apidevtools/swagger-parser';
+import cloneDeep from 'lodash/cloneDeep';
+import merge from 'lodash/merge';
+import { OpenAPI } from 'openapi-types';
+import { addExtension,
+  isSchemaObject,
+  OpenAPIObject,
+  SchemaObject } from 'openapi3-ts';
+
+const normalizeArray = (arr: unknown): unknown[] => Array.isArray(arr) ? arr : [
+  arr,
+];
+
+const objectify = (thing: unknown): Record<any, any> => {
+  if (!(!!thing && typeof thing === 'object')) {
+    return {
+    };
+  }
+
+  return thing as Record<any, any>;
+};
+
+const primitives = {
+  /* eslint-disable @typescript-eslint/camelcase */
+  boolean: (
+    schema: SchemaObject
+  ): boolean => typeof schema.default === 'boolean' ? schema.default : true,
+  integer: (): number => 0,
+  number: (): number => 0,
+  number_float: (): number => 0.0,
+  string: (): string => 'string',
+  string_date: (): string => new Date().toISOString().substring(0, 10),
+  'string_date-time': (): string => new Date().toISOString(),
+  string_email: (): string => 'user@example.com',
+  string_hostname: (): string => 'example.com',
+  string_ipv4: (): string => '198.51.100.42',
+  string_ipv6: (): string => '2001:0db8:5b96:0000:0000:426f:8e17:642a',
+  string_uuid: (): string => '3fa85f64-5717-4562-b3fc-2c963f66afa6',
+  /* eslint-enable @typescript-eslint/camelcase */
+};
+
+const primitive = (schema: SchemaObject): unknown => {
+  schema = objectify(schema);
+  const { type, format } = schema;
+
+  const fn = (primitives as any)[`${type}_${format}`]
+    || (primitives as any)[(type as string)];
+
+  if (typeof(fn) === 'function') {
+    return fn(schema);
+  }
+
+  return 'Unknown Type: ' + schema.type;
+};
+
+const deeplyStripKey = (
+  input: any,
+  keyToStrip: string,
+  predicate: (
+    a: any,
+    b: any
+  ) => boolean = (): boolean => true
+): Record<any, any> => {
+  if (
+    typeof input !== 'object'
+      || Array.isArray(input)
+      || input === null
+      || !keyToStrip
+  ) {
+    return input;
+  }
+
+  const obj = Object.assign({
+  }, input);
+
+  Object.keys(obj).forEach(k => {
+    if (k === keyToStrip && predicate(obj[k], k)) {
+      delete obj[k];
+      return;
+    }
+    obj[k] = deeplyStripKey(obj[k], keyToStrip, predicate);
+  });
+
+  return obj;
+};
+
+const sampleFromSchema = (schema: SchemaObject, config: any = {
+}): any => {
+  let { type } = schema;
+  const {
+    allOf,
+    additionalProperties,
+    example,
+    items,
+    properties,
+  } = schema;
+  const { includeReadOnly, includeWriteOnly } = config;
+
+  if (example !== undefined) {
+    return deeplyStripKey(example, '$$ref', (val: unknown) => {
+      // do a couple of quick sanity tests to ensure the value
+      // looks like a $$ref that swagger-client generates.
+      return typeof val === 'string' && val.indexOf('#') > -1;
+    });
+  }
+
+  if (allOf) {
+    const foo = {
```

Comment:
What's the purpose of foo?

Reply:
Ah, I just forgot to think of a proper name for it. I'll rename it to `composedSchema`. The `OpenApi`  spec allows for Schema composition by using `allOf` and then listing an array of multiple schemas.